[SPARK-27924][SQL][TEST][FOLLOW-UP] Enable Boolean-Predicate syntax tests

beliefer opened the pull request:
## What changes were proposed in this pull request?

This PR is a follow-up to https://github.com/apache/spark/pull/25074 

## How was this patch tested?

Pass the Jenkins with the newly update test files.

beliefer committed:
test

beliefer commented:
@dongjoon-hyun Could you take a look at this PR?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13787/
Test PASSed.

SparkQA commented:
**[Test build #108702 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108702/testReport)** for PR 25366 at commit [`9654246`](https://github.com/apache/spark/commit/965424655dbe8bfdb5a9b162724b270bfedf5cbe).

beliefer committed:
update test output.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13788/
Test PASSed.

SparkQA commented:
**[Test build #108703 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108703/testReport)** for PR 25366 at commit [`05a14c5`](https://github.com/apache/spark/commit/05a14c54ea8b5496bf473f3001b5c7ce028e75e9).

wangyum commented on sql/core/src/test/resources/sql-tests/results/pgSQL/boolean.sql.out:
```diff
@@ -566,176 +566,332 @@ struct<tf_12_ff_4:string,f1:boolean,f1:boolean>
 
 -- !query 62
-INSERT INTO BOOLTBL3 VALUES ('true', true, 1)
+SELECT '' AS "Not False", f1
```
`"Not False"` -> ``` `Not False` ```?

beliefer replied:
OK. I will change it. Thanks @wangyum .

beliefer committed:
Fix bug.

SparkQA commented:
**[Test build #108705 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108705/testReport)** for PR 25366 at commit [`5090742`](https://github.com/apache/spark/commit/509074200f0052dbf9da0ee74c3bceb252ef388c).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13790/
Test PASSed.

SparkQA commented:
**[Test build #108702 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108702/testReport)** for PR 25366 at commit [`9654246`](https://github.com/apache/spark/commit/965424655dbe8bfdb5a9b162724b270bfedf5cbe).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108702/
Test FAILed.

beliefer committed:
generate a new result file.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13797/
Test PASSed.

SparkQA commented:
**[Test build #108712 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108712/testReport)** for PR 25366 at commit [`7278128`](https://github.com/apache/spark/commit/7278128886ba36deab47b2a5b8c72bec937e711d).

SparkQA commented:
**[Test build #108703 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108703/testReport)** for PR 25366 at commit [`05a14c5`](https://github.com/apache/spark/commit/05a14c54ea8b5496bf473f3001b5c7ce028e75e9).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108703/
Test PASSed.

SparkQA commented:
**[Test build #108705 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108705/testReport)** for PR 25366 at commit [`5090742`](https://github.com/apache/spark/commit/509074200f0052dbf9da0ee74c3bceb252ef388c).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108705/
Test PASSed.

SparkQA commented:
**[Test build #108712 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108712/testReport)** for PR 25366 at commit [`7278128`](https://github.com/apache/spark/commit/7278128886ba36deab47b2a5b8c72bec937e711d).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108712/
Test FAILed.

dongjoon-hyun commented:
Sure, @beliefer . I'll take a look.  Thank you for pinging me.

dongjoon-hyun commented:
@beliefer . Please regenerate the result. That will recover the UT failures.

dongjoon-hyun commented:
BTW, @beliefer . This had better be a followup of yours ([SPARK-27924][SQL] Support ANSI SQL Boolean-Predicate syntax). I updated the PR title. Also, I changed `How was this patch tested?` section. Since you updated the test cases which are verified by Jenkins. It should not be 'N/A'.

dongjoon-hyun commented:
I made a PR to you, @beliefer . Please review and merge.
- https://github.com/beliefer/spark/pull/2

maropu commented:
Does https://github.com/apache/spark/pull/25357 afffects the output in this pr? If so, I think we'd be better to merge that pr first to make code changes less.

beliefer commented:
> BTW, @beliefer . This had better be a followup of yours ([SPARK-27924][SQL] Support ANSI SQL Boolean-Predicate syntax). I updated the PR title. Also, I changed `How was this patch tested?` section. Since you updated the test cases which are verified by Jenkins. It should not be 'N/A'.

Thank you very much!

beliefer commented:
> I made a PR to you, @beliefer . Please review and merge.
> 
> * [beliefer#2](https://github.com/beliefer/spark/pull/2)

Thanks for your help! It's my mistake. I forget something.

beliefer committed:
regenerate new output file.

SparkQA commented:
**[Test build #108741 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108741/testReport)** for PR 25366 at commit [`30ed8d3`](https://github.com/apache/spark/commit/30ed8d3f61e5455d17efe77e6e5985c7e72d0109).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13822/
Test PASSed.

SparkQA commented:
**[Test build #108741 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108741/testReport)** for PR 25366 at commit [`30ed8d3`](https://github.com/apache/spark/commit/30ed8d3f61e5455d17efe77e6e5985c7e72d0109).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108741/
Test PASSed.

dongjoon-hyun commented:
@maropu . IMO, we need more review on #25357 ~ I'll comment on that PR.

dongjoon-hyun reviewed (approved):
@beliefer , @maropu , @wangyum .
I'll merge this first. 

+1, LGTM. Merged to master. Thank you, @beliefer .

beliefer commented:
@dongjoon-hyun  Thanks for your help, review and work!
